OutOfSchool/OutOfSchool.WebApi.Tests/Services/Database/WorkshopServiceDBTests.cs

```diff
@@ -0,0 +1,191 @@
+    private Mock<IProviderRepository> providerRepositoryMock;
+
+    [SetUp]
+    public async Task SetUp()
+    {
+        dbContextOptions = new DbContextOptionsBuilder<OutOfSchoolDbContext>()
+            .UseInMemoryDatabase(databaseName: "OutOfSchoolTestDB")
+            .UseLazyLoadingProxies()
+            .EnableSensitiveDataLogging()
+            .Options;
+
+        dbContext = new OutOfSchoolDbContext(dbContextOptions);
+
+        workshopRepository = new WorkshopRepository(dbContext);
+        dateTimeRangeRepository = new Mock<IEntityRepositorySoftDeleted<long, DateTimeRange>>();
+        roomRepository = new Mock<IEntityRepositorySoftDeleted<Guid, ChatRoomWorkshop>>();
+        teacherService = new Mock<ITeacherService>();
+        logger = new Mock<ILogger<WorkshopService>>();
+        workshopImagesMediator = new Mock<IImageDependentEntityImagesInteractionService<Workshop>>();
+        mapper = TestHelper.CreateMapperInstanceOfProfileTypes<CommonProfile, MappingProfile>();
+        providerAdminRepository = new Mock<IProviderAdminRepository>();
+        averageRatingServiceMock = new Mock<IAverageRatingService>();
+        providerRepositoryMock = new Mock<IProviderRepository>();
+
+        workshopService =
+            new WorkshopService(
+                workshopRepository,
+                dateTimeRangeRepository.Object,
+                roomRepository.Object,
+                teacherService.Object,
+                logger.Object,
+                mapper,
+                workshopImagesMediator.Object,
+                providerAdminRepository.Object,
+                averageRatingServiceMock.Object,
+                providerRepositoryMock.Object);
+
+        Seed();
+    }
+
+    [TearDown]
+    public void Dispose()
+    {
+        dbContext.Dispose();
+    }
+
+    [Test]
+    public async Task GetByFilter_WhenSetFormOfLearning_ShouldBuildPredicateAndReturnEntities()
+    {
+        // Arrange
+        await SeedFormOfLearningWorkshops();
+
+        var filter = new WorkshopFilter()
+        {
+            FormOfLearning = new List<FormOfLearning>()
+            {
+                FormOfLearning.Offline,
+            },
+        };
+
+        // Act
+        var result = await workshopService.GetByFilter(filter).ConfigureAwait(false);
+
+        // Assert
+        Assert.AreEqual(2, result.TotalAmount);
+    }
+
+    [Test]
+    public async Task GetByFilter_WhenSetFormOfLearning_ReturnNothing()
+    {
+        // Arrange
+        await SeedFormOfLearningWorkshops();
+
+        var filter = new WorkshopFilter()
+        {
+            FormOfLearning = new List<FormOfLearning>()
+            {
+                FormOfLearning.Mixed,
+            },
+        };
+
+        // Act
+        var result = await workshopService.GetByFilter(filter).ConfigureAwait(false);
+
+        // Assert
+        Assert.AreEqual(0, result.TotalAmount);
+    }
+
+    [Test]
+    public async Task GetByFilter_WhenSetFormOfLearning_ReturnAll()
```

Comment:
(Test naming) Probably this test should be named `...WhenNotSetFormOfLearning...`?

Reply:
Thanks. Renamed.

Reply:
Yes it was, and no it doesn't meat expectations ) 
And sorry, I wasn't specific enough. `GetByFilter_WhenNotSetFormOfLearning_ReturnAll` was the proposed name, not the `GetByFilter_WhenSetFormOfLearning_WhenNotSetFormOfLearning`

Please, be more gentle to you names (and subsequent readers)

Reply:
Ok. Sorry. Renamed.